core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1290,7 +1291,7 @@ class DAGScheduler(
       case TaskResultLost =>
         // Do nothing here; the TaskScheduler handles these failures and resubmits the task.
 
-      case other =>
+      case _: ExecutorLostFailure | TaskKilled | UnknownReason =>
```

Comment:
this was needed just for the added test case -- I needed a way to intentionally throw an error inside the event loop, and with this change if `event.reason` is `null`, you get an exception.

And yeah, if you add another type but dont' include it in this match, you'll get a fatal warning in compilation since its a sealed trait.